app/code/Magento/PageCache/Model/App/FrontController/BuiltinPlugin.php

```diff
@@ -89,9 +89,12 @@ public function aroundDispatch(
      */
     protected function addDebugHeaders(ResponseHttp $result)
     {
-        $cacheControl = $result->getHeader('Cache-Control')->getFieldValue();
-        $this->addDebugHeader($result, 'X-Magento-Cache-Control', $cacheControl);
-        $this->addDebugHeader($result, 'X-Magento-Cache-Debug', 'MISS', true);
+        $cacheControlHeader = $result->getHeader('Cache-Control');
+        if ($cacheControlHeader instanceof \Zend\Http\Header\HeaderInterface) {
+            $cacheControl = $cacheControlHeader->getFieldValue();
+            $this->addDebugHeader($result, 'X-Magento-Cache-Control', $cacheControl);
+            $this->addDebugHeader($result, 'X-Magento-Cache-Debug', 'MISS', true);
```

Comment:
I agree. I have changed the code like that. (not sure why the test is failed for PHP 5.5, but it doesn't related to this fix)